client/my-sites/stats/stats-date-picker/index.jsx

```diff
@@ -51,6 +56,21 @@ class StatsDatePicker extends Component {
 		}
 	}
 
+	dateForCustomRange() {
+		const { query, moment, translate } = this.props;
+
+		const localizedStartDate = moment( query.start_date );
+		const localizedEndDate = moment( query.date );
+
+		return translate( '%(startDate)s ~ %(endDate)s', {
```

Comment:
We should be able to leverage the changes from @annacmc which just got merged

https://github.com/Automattic/wp-calypso/pull/96661/files#diff-7b415fb32f446cf84f379854d3e2a15a55564aae14c3f9eb1f5cf404bdb910bfL55